pydantic/types.py

```diff
@@ -261,14 +262,18 @@ class ConstrainedInt(int, metaclass=ConstrainedNumberMeta):
 
     @classmethod
     def __get_validators__(cls) -> 'CallableGenerator':
+        if cls.strict:
+            yield make_arbitrary_type_validator(int)
```

Comment:
Should `StrictInt` allow `bool`? I think with the current behaviour it will. I think it shouldn't

You might need a custom method here.

Reply:
You are right, bool is a subclass of int. I will create a more specific validation method for this.